lib/internal/per_context/primordials.js

```diff
@@ -21,10 +21,28 @@ const {
 // `uncurryThis` is equivalent to `func => Function.prototype.call.bind(func)`.
 // It is using `bind.bind(call)` to avoid using `Function.prototype.bind`
 // and `Function.prototype.call` after it may have been mutated by users.
-const { bind, call } = Function.prototype;
+const { apply, bind, call } = Function.prototype;
 const uncurryThis = bind.bind(call);
 primordials.uncurryThis = uncurryThis;
 
+const applyBind = bind.bind(apply);
```

Comment:
Could you please add a comment like the `uncurryThis` one which explains this is equivalent to
```js
const applyBind = (func) => (thisArg, args) => ReflectApply(func, thisArg, args);
```

Reply:
Actually, it’s closer to:
```js
const applyBind = (func, ...rest) => {
	if (args.length > 0) {
		const thisArg = rest[0];
		return (args = []) => ReflectApply(func, thisArg, args);
	} else {
		return (thisArg, args = []) => ReflectApply(func, thisArg, args);
	}
}

Reply:
BTW, shouldn't we prefer using `Reflect.apply` instead of `Function.prototype.apply`? I think if someone calls those methods without a argument array, it's almost certainly a bug, and using `Reflect.apply` would help spot that. Or is there a performance reason to use one or the other?

Reply:
Well, `Reflect.apply` isn’t very optimised in comparison to bound `Function.prototype.call` and `Function.prototype.apply` calls.

Also, engines are able to generate better code for native bound functions compared to arrow functions that replicate the behaviour of a bound function.

Reply:
I meant to use `const { apply }  = Reflect` compared to `const { apply } = Function.prototype`.

Reply:
That wouldn’t work, since then `applyBind(foo)` would create a function that roughly does: `(...args) => Reflect.apply.call(foo, ...args)`, instead of `(...args) => Reflect.apply(foo, ...args)`.

Reply:
Gotcha, the equivalent would have to be:
```suggestion
const applyBind = bind.bind(Reflect.apply, null);
```

> `Reflect.apply` isn’t very optimised in comparison to bound `Function.prototype.call` and `Function.prototype.apply` calls.

Is that generally true statement or is it specific to V8? When reading the spec, `Reflect.apply` seems to be doing less things than `Function.prototype.apply`, that's a bit counter-intuitive.

Reply:
> > `Reflect.apply` isn’t very optimised in comparison to bound `Function.prototype.call` and `Function.prototype.apply` calls.
> 
> Is that generally true statement or is it specific to V8?

That’s mostly specific to **V8**, since **V8** hasn’t put as much effort into optimising `Reflect` as into optimising `Object` and `Function` methods.

**Refs:** <https://github.com/nodejs/node/pull/36740#issuecomment-753545365>